lib/repl.js

```diff
@@ -319,9 +320,16 @@ function REPLServer(prompt,
   self.outputStream = output;
 
   self.resetContext();
-  self.bufferedCommand = '';
   self.lines.level = [];
 
+  self.clearBufferedCommand();
+  Object.defineProperty(this, 'bufferedCommand', {
+    get: util.deprecate(() => self[kBufferedCommandSymbol],
+                        'REPLServer.bufferedCommand is deprecated'),
+    set: util.deprecate((val) => self[kBufferedCommandSymbol] = val,
+                        'REPLServer.bufferedCommand is deprecated')
```

Comment:
This would need an assigned deprecation code. See `doc/api/deprecations.md`.
Before this lands, make the code `DEP00XX` and whom ever does the landing would assign the actual code.
The code would be passed in as the third argument for the `util.deprecate()` method.